packages/gatsby/src/internal-plugins/dev-404-page/raw_dev-404-page.js

```diff
@@ -1,6 +1,7 @@
 import React from "react"
 import PropTypes from "prop-types"
-import { graphql, Link } from "gatsby"
+import { graphql, Link, navigate } from "gatsby"
+import queryString from "query-string"
```

Comment:
This package need to be added to dependencies of `gatsby`

Reply:
It somewhat works right now only because it's transitive dependency of other things - but we should explicitly declare it as dependency - this in practice will likely break yarn PnP and can break regular npm / yarn installations if `node_modules` will be created in unusual ways

Reply:
I prefer `query-string` rather than `qs`, and both are already installed as transitive dependencies, so I could either use one or the other. If there's no preference, I'll install `query-string`

Reply:
`query-string` is fine - we just need to add it to `dependencies` in https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/package.json

Reply:
> this in practice will likely break yarn PnP and can break regular npm / yarn installations if node_modules will be created in unusual ways

after re-reading it I figured it might be confusing - I meant not adding package to dependencies would cause problems I listed